build.gradle

```
@@ -535,6 +541,24 @@ configure(subprojects - getJsProjects()) { subproject ->
     }
 }
 
+def checkDexer = task('checkDexer', description: 'compiles android class files into dex bytecode', dependsOn: compileKotlin) {
```

as you test on the jar (which makes more sense than on the class files) you should change the dependsOn accordingly:
```suggestion
def checkDexer = task('checkDexer', description: 'compiles android class files into dex bytecode', dependsOn: jar) {
```